File: examples/performance/primes/primes.rb
SIZE = 1_000_000
known = [2, 3]
n = 5
i = 2
j = 1
limit = Math.sqrt(n)
while i < SIZE do
    isPrime = true
    j = 1
    while isPrime && known[j] < limit && j < i do
        if n % known[j] == 0 then
            isPrime = false
        end
        j += 1
    end
    if isPrime then
        known[i] = n
        i += 1
        puts n
    end
    n += 2
    limit = Math.sqrt(n)
end

File: examples/performance/primes/primes.go
// gcc primes.c -lm -o primes

package main

import "fmt"
import "math"

const SIZE = 1000000

func main() {
    var known [SIZE]int
    known [0] = 2
    known [1] = 3

    var n = 5
    var i = 2
    var j = 1
    var limit = math.Sqrt(float64(n))

    for ; i < SIZE ; {
        var isPrime = true

        for j = 1; isPrime && float64(known[j]) < limit && j < i; j++ {
            if n % known[j] == 0 {
                isPrime = false
            }
        }

        if isPrime {
            known[i] = n
            i++

            fmt.Println(n)
        }

        n += 2
        limit = math.Sqrt(float64(n))
    }

    return
}
